src/theory/arith/nl/ext/arith_nl_compare_proof_gen.cpp

```diff
@@ -0,0 +1,339 @@
+      Node v = isDisequalZero(e);
+      Assert (!v.isNull());
+      deq[v] = e;
+      continue;
+    }
+    expc.emplace_back(ec);
+    // a comparsion literal
+    if (e != ec)
+    {
+      Node eeq = e.eqNode(ec);
+      Node eeqSym = ec.eqNode(e);
+      cdp.addTrustedStep(
+          eeqSym, TrustId::ARITH_NL_COMPARE_LIT_TRANSFORM, {}, {});
+      cdp.addStep(ec, ProofRule::EQ_RESOLVE, {e, eeq}, {});
+    }
+    // add to product
+    Assert(ec.getNumChildren() == 2);
+  }
+  // reorder the explanation based on the order it appears in the conclusion
+  Node concc = getCompareLit(conc);
+  Assert(!concc.isNull());
+  Assert(concc.getNumChildren() == 2);
+  Assert (concc[0].getKind() == Kind::ABS);
+  std::vector<Node> cprod[2];
+  Kind ck = decomposeCompareLit(concc, cprod[0], cprod[1]);
+  // convert to exponent counts
+  std::map<Node, size_t> mexp[2];
+  for (size_t i = 0; i < 2; i++)
+  {
+    std::vector<Node>& cpi = cprod[i];
+    std::map<Node, size_t>& mi = mexp[i];
+    for (const Node& p : cpi)
+    {
+      mi[p]++;
+    }
+  }
+  // if GT conclusion, ensure the first explanation is GT, which makes checking
+  // simpler
+  if (ck==Kind::GT)
+  {
+    for (size_t i=0, nexp = expc.size(); i<nexp; i++)
+    {
+      if (expc[i].getKind()==Kind::GT)
+      {
+        if (i>0)
+        {
+          Node tmp = expc[i];
+          expc[i] = expc[0];
+          expc[0] = tmp;
+        }
+        break;
+      }
+    }
+  }
+  std::vector<size_t> eexp;
+  // reorder the conclusion based on the explanation
+  NodeManager* nm = nodeManager();
+  std::vector<Node> cprodt[2];
+  for (const Node& e : expc)
+  {
+    std::vector<Node> eprod[2];
+    decomposeCompareLit(e, eprod[0], eprod[1]);
+    Assert(eprod[0].size() <= 1 && eprod[1].size() <= 1);
+    for (size_t i = 0; i < 2; i++)
+    {
+      if (eprod[i].empty())
+      {
+        size_t ii = 1 - i;
+        Node a = eprod[ii][0];
+        size_t na = mexp[ii][a];
+        size_t nb = mexp[i][a];
+        // Don't take more than this side has. This handles cases like
+        // (> (abs x) (abs 1)) => (> (abs (* x x)) (abs x)),
+        // where we should only consume one copy of x.
+        size_t n = na-nb;
+        Node one = mkOne(a.getType());
+        for (size_t j=0; j<n; j++)
+        {
+          cprodt[i].push_back(one);
+          cprodt[ii].push_back(a);
+        }
+        mexp[ii][a] -= n;
+        eexp.emplace_back(n);
+        break;
+      }
+      else if (i == 1)
+      {
+        // both non-empty, take min
+        Node a = eprod[0][0];
+        Node b = eprod[1][0];
+        size_t na = mexp[0][a];
+        size_t nb = mexp[1][b];
+        size_t n = na > nb ? nb : na;
+        for (size_t j = 0; j < 2; j++)
+        {
+          const Node& c = eprod[j][0];
+          for (size_t k=0; k<n; k++)
+          {
+            cprodt[j].push_back(c);
+          }
+          mexp[j][c] -= n;
+        }
+        eexp.emplace_back(n);
+      }
```

Comment:
Good point, although there is a hidden restriction in the current implementation, the above lemma would not happen since we compare factors in order so that all comparisons with a factor occur in consecutive premises (where in your example `a` does not).

I will add a comment about this.